sdk/src/account_data.rs

```diff
@@ -0,0 +1,103 @@
+#![cfg(feature = "full")]
+use serde::de::{Deserialize, Deserializer};
+use serde::ser::{Serialize, Serializer};
+use std::{
+    ops::{Deref, DerefMut, Index, IndexMut},
+    slice::SliceIndex,
+};
+
+#[derive(Debug, Default)]
+pub struct AccountData {
+    data: Vec<u8>,
+}
+
```

Comment:
Per our previous discussions, I will integrate https://github.com/solana-labs/solana/pull/15756 
once you are done with your changes. It will be easier if you just keep `data` in `AccountSharedData` as `Arc<Vec<u8>>` as in your initial plans and what we discussed.

Can you please remove this `account_data.rs` from this pull request?